apps/builder/app/builder/features/breakpoints/width-input.tsx

```diff
@@ -55,15 +50,21 @@ const useEnhancedInput = ({
     onChangeComplete: handleChangeComplete,
   });
 
+  const getValue = () => {
+    const value = inputRef.current?.valueAsNumber;
+    return typeof value !== "number" || isNaN(value) ? min : value;
+  };
+
   return {
     ref: scrubRef,
     inputRef,
-    onChange(event: ChangeEvent<HTMLInputElement>) {
-      setIntermediateValue(event.target.valueAsNumber);
+    onChange() {
+      setIntermediateValue(getValue());
```

Comment:
How does it work? Does input get the state before it is updated by react?

Reply:
its a controlled input, so on change you can pretty much get the value from the element

Reply:
though I want to refactor this again, because currently there is no invalid state handling so you can't just delete the value to type a new one